tools/firmware.toit

```diff
@@ -1357,13 +1350,97 @@ class Envelope:
       throw "cannot open envelope - expected version $ENVELOPE-FORMAT-VERSION, was $version"
     return version
 
+class RelocationInformation:
+  relocation-bytes/ByteArray
+  word-size/int  // In bytes.
+
+  constructor.from relocatable/ByteArray --.word-size:
+    chunk-size := get-relocatable-chunk-byte-size --word-size=word-size
+    chunk-count := ceil_ relocatable.size chunk-size
+    relocation-bytes = ByteArray chunk-count * word-size
+    for i := 0; i < chunk-count; i++:
+      relocation-bytes.replace (i * word-size) relocatable[i * chunk-size..i * chunk-size + word-size]
+
+  make-relocatable image/ByteArray -> ByteArray:
+    relocatable-chunk-size := get-relocatable-chunk-byte-size --word-size=word-size
+    relocated-chunk-size := get-relocated-chunk-byte-size --word-size=word-size
+    result := ByteArray (ceil_ image.size relocated-chunk-size) * relocatable-chunk-size
+
+    image-i := 0
```

Comment:
I think it confuses me a bit that these are called `-i`. That sounds like an index of sorts, but here it is very clearly offsets.